scripts/initial_release_prep.py

```diff
@@ -0,0 +1,312 @@
+        'The general instructions are as follows:\n\n'
+        '1. Login as admin\n'
+        '2. Navigate to the admin panel and then the jobs tab\n'
+        '3. Run the above jobs\n'
+        '4. In case of failure/success, please send the output logs for '
+        'the job to me and the job author:\n\n'
+        'Thanks!\n') % job_names_with_instructions_and_author
+    return mail_message_template
+
+
+def get_extra_jobs_due_to_schema_changes(
+        remote_alias, previous_release_version):
+    """Finds additional jobs which should be run based on
+    schema changes in feconf.
+
+    Args:
+        remote_alias: str. The alias for Oppia repo.
+        previous_release_version: str. The version of the previous release.
+
+    Returns:
+        list(str). The list of jobs to run based on schema changes.
+    """
+    schema_versions_to_jobs_mapping = {
+        'CURRENT_COLLECTION_SCHEMA_VERSION': 'CollectionMigrationOneOffJob',
+        'CURRENT_STATE_SCHEMA_VERSION': 'ExplorationMigrationJobManager',
+        'CURRENT_SKILL_CONTENTS_SCHEMA_VERSION': 'SkillMigrationOneOffJob',
+        'CURRENT_MISCONCEPTIONS_SCHEMA_VERSION': 'SkillMigrationOneOffJob',
+        'CURRENT_RUBRIC_SCHEMA_VERSION': 'SkillMigrationOneOffJob',
+        'CURRENT_STORY_CONTENTS_SCHEMA_VERSION': 'StoryMigrationOneOffJob',
+        'CURRENT_SUBTOPIC_SCHEMA_VERSION': 'TopicMigrationOneOffJob',
+        'CURRENT_STORY_REFERENCE_SCHEMA_VERSION': 'TopicMigrationOneOffJob'}
+
+    diff_output = common.run_cmd([
+        'git', 'diff', '%s/develop' % remote_alias,
+        '%s/release-%s' % (remote_alias, previous_release_version),
+        'feconf.py'])
+    feconf_diff = diff_output[:-1].split('\n')
+
+    jobs_to_run = []
+    for version_key in schema_versions_to_jobs_mapping:
+        for line in feconf_diff:
+            if line.startswith(('+%s' % version_key, '-%s' % version_key)):
+                jobs_to_run.append(schema_versions_to_jobs_mapping[version_key])
+
+    return list(set(jobs_to_run))
+
+
+def check_changes_in_supported_audio_languages(
+        remote_alias, previous_release_version):
+    """Checks changes in constants.SUPPORTED_AUDIO_LANGUAGES between
+    the current and previous release.
+
+    Args:
+        remote_alias: str. The alias for Oppia repo.
+        previous_release_version: str. The version of the previous release.
+
+    Returns:
+        bool. Whether supported audio languages have changed.
+    """
+    try:
+        from constants import constants
+        supported_audio_languages_for_current_release = constants[
+            'SUPPORTED_AUDIO_LANGUAGES']
+
+        common.run_cmd([
+            'git', 'checkout',
+            '%s/release-%s' % (remote_alias, previous_release_version),
+            '--', 'assets/constants.ts'])
+        from constants import constants
+        supported_audio_languages_for_previous_release = constants[
+            'SUPPORTED_AUDIO_LANGUAGES']
+    finally:
```

Comment:
What's the try block doing here? And why do we expect an exception?

Reply:
The command [here](https://github.com/oppia/oppia/pull/8005/files/60b8dad47d0d646f925dce41ea75a405c9d925c3#diff-5ec19cb5334ecdb7f1dada588baf5b5eR182) converts the constants file to a version from previous release and stages it as well (`git add`). So, the try finally is added to be safe in case of any random exception i.e. the changes to constants file are reverted even in case the function faces an exception.

Reply:
Ah I see.. Have you run into an exception here while testing it?

Reply:
No, I did not get any specific exception while testing.